src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -2179,129 +2172,43 @@ private boolean areAllMedicalsInInventory() throws OHServiceException {
-					}
-				} catch (OHServiceException e) {
-					OHServiceExceptionUtil.showMessages(e);
-				}
-			}
-		});
-
-		actions.put(radioWithMovement, () -> {
-			if (medicalTypeSelected.getDescription().equals(MessageBundle.getMessage("angal.common.all.txt"))) {
-				try {
-					if (!areAllMedicalsInInventory()) {
-						int info = (!inventoryRowSearchList.isEmpty())
-							? MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttoaddallnotyetlistedproducts.msg")
-							: JOptionPane.YES_OPTION;
-						if (info == JOptionPane.YES_OPTION) {
-							jTableInventoryRow.setModel(new InventoryRowModel(null, true));
-							fireInventoryUpdated();
-							if (!inventoryRowList.isEmpty()) {
-								MessageDialog.info(null, "angal.invetory.tablehasbeenupdated.msg");
-							}
-						}
-					} else {
-						MessageDialog.info(null, "angal.inventory.youhavealreadyaddedallproduct.msg");
-					}
-				} catch (OHServiceException e) {
-					OHServiceExceptionUtil.showMessages(e);
-				}
-			} else {
-				try {
-					if (!areAllMedicalsInInventory()) {
-						int info = (!inventoryRowSearchList.isEmpty())
-							? MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttoaddallnotyetlistedproducts.msg")
-							: JOptionPane.YES_OPTION;
-						if (info == JOptionPane.YES_OPTION) {
-							MedicalType medType = (MedicalType) medicalTypeComboBox.getSelectedItem();
-							jTableInventoryRow.setModel(new InventoryRowModel(medType, true));
-							fireInventoryUpdated();
-							if (!inventoryRowList.isEmpty()) {
-								MessageDialog.info(null, "angal.invetory.tablehasbeenupdated.msg");
-							}
-						}
-					} else {
-						MessageDialog.info(null, "angal.inventory.youhavealreadyaddedallproduct.msg");
-					}
-				} catch (OHServiceException e) {
-					OHServiceExceptionUtil.showMessages(e);
-				}
-			}
-		});
+	    actions.put(radioButtonAll, () -> handleInventoryUpdate(true, false, false));
+	    actions.put(radioOnlyNonZero, () -> handleInventoryUpdate(false, false, true));
+	    actions.put(radioWithMovement, () -> handleInventoryUpdate(false, true, false));
+	    noData = false;
+	}
+
+	private void handleInventoryUpdate(boolean includeAll, boolean withMovement, boolean nonZero) {
+	    boolean isAllSelected = medicalTypeSelected.getDescription().equals(MessageBundle.getMessage("angal.common.all.txt"));
+	    MedicalType medType = isAllSelected ? null : (MedicalType) medicalTypeComboBox.getSelectedItem();
+
+	    try {
+	        if (!areAllMedicalsInInventory()) {
+	            int userChoice = (!inventoryRowSearchList.isEmpty()) 
+	                ? MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttoaddallnotyetlistedproducts.msg") 
+	                : JOptionPane.YES_OPTION;
+
+	            if (userChoice == JOptionPane.YES_OPTION) {
+	                jTableInventoryRow.setModel(new InventoryRowModel(medType, nonZero,includeAll, withMovement));
+	                fireInventoryUpdated();
+	                adjustWidth();
```

Comment:
In particular, this method has been removed, because by introducing `jTableInventoryRow.setAutoCreateColumnsFromModel(false)`, after first JTable creation we don't have to worry about the appearence every time we update the model

Reply:
you talk about adjustWidth(); rigth? I have already remove it.

Reply:
my comment was before your removal... ok, thanks